src/test/python/agents/script/sample_script.py

```diff
@@ -0,0 +1,20 @@
+import random
+from time import sleep
+
+from rlbot.agents.base_script import BaseScript
+
+
+class SampleScript(BaseScript):
+    def run(self):
+        for i in range(10):
+            packet = self.get_game_tick_packet()
+            self.renderer.begin_rendering()
+            self.renderer.draw_string_2d(random.randint(200, 300), random.randint(20, 300), 1, 1,
+                                         str(packet.game_info.seconds_elapsed), self.renderer.yellow())
+            self.renderer.end_rendering()
+            sleep(5)
+
+
+print("Sample script starting...")
+script = SampleScript("myScript")
+script.run()
```

Comment:
Please document when things get loaded. 
I think it's dangerous to do state-changing things at import time (prevents reuse) and I'd recommend having a callback (e.g. `on_added_to_match`) instead.

Reply:
Would it be adequate for the user to have one of these?
```python
if __name__ == '__main__':
    do_whatever()
```

I'd prefer to avoid the callback pattern because that would add more complexity and constraints I think:
- The scripts must extend a base class (currently it's purely optional)
- The framework needs to do class loading for the scripts, which has historically been tricky

Reply:
- I'm note sure if the `if __name__ == '__main__'` helps here as you should only get `'__main__'` back if you're being run at top level and are expected to do things like command line flag parsing. It's possible to import it with a custom name but that doesn't appeal to me.
- I was thinking that `on_added_to_match()` could just be functions like how RLBotTraining does it with `make_default_playlist()` [[1]](https://github.com/RLBot/RLBotTraining/blob/580e162e54f41d278319bf4f68d99dab7f3b7ed8/rlbottraining/exercise_runner.py#L137) [[2]](https://github.com/RLBot/RLBotTraining/blob/580e162e54f41d278319bf4f68d99dab7f3b7ed8/rlbottraining/common_exercises/bronze_striker.py#L79)
- We seem to be doing module loading already.

Reply:
`if __name__ == '__main__'` is working as intended in https://github.com/tarehart/RLBotTwitchBroker/blob/master/tst/boost_monkey.py#L99-L101

I may be misunderstanding you, but to be clear the scripts are started with `subprocess.Popen([sys.executable, script_config_bundle.script_file]...`. For scripts we're not doing any module loading yet, and they get run in a 'top level' manner.

Your `make_default_playlist()` does seem nice and optional and lightweight, but I don't see it adding much value in this case since `if __name__ == '__main__'` is working as intended.

Reply:
:+1: I missed the `subprocess.Popen()`.